api/types/types.proto

```diff
@@ -583,6 +572,48 @@ message AuditConfig {
     double WriteTargetValue = 13 [ (gogoproto.jsontag) = "write_target_value,omitempty" ];
 }
 
+// ClusterNetworkingConfigV2 implements the ClusterNetworkingConfig interface.
+message ClusterNetworkingConfigV2 {
```

Comment:
should be `ClusterNetworkingConfigV1`
versioning is per-resource, not API-wide

Reply:
Not sure about that, there seems to be an API-wide notion of ["V2 style objects"](https://github.com/gravitational/teleport/blob/b1fd129b225165fa82f29201eb26a8fec3efc91e/lib/services/resource.go#L561-L562) (that has to do with the kind/spec/metadata/version top-level structuring of resource objects?).

Reply:
In the code base as it is, `ClusterNetworkingConfigV2` makes more sense to me as well. However without that context, it would make more sense to start objects as `V1`, so maybe we should try to abandon this notion of "V2 style objects".  We could update the `V2SchemaTemplate` comment to be more accurate, as it's just the second version of schema templating and is used in V3+ as well.